CI new numpy nightly failing some tests

```
2022-12-15T09:04:34.1328103Z FAILED pandas/tests/series/methods/test_describe.py::TestSeriesDescribe::test_numeric_result_dtype[complex] - TypeError: a must be an array of real numbers
2022-12-15T09:04:34.1328576Z FAILED pandas/tests/series/methods/test_describe.py::TestSeriesDescribe::test_numeric_result_dtype[complex64] - TypeError: a must be an array of real numbers
2022-12-15T09:04:34.1329046Z FAILED pandas/tests/series/methods/test_describe.py::TestSeriesDescribe::test_numeric_result_dtype[complex128] - TypeError: a must be an array of real numbers
```

Can reproduce this locally with:
```
pip uninstall numpy -y
pip install -i https://pypi.anaconda.org/scipy-wheels-nightly/simple numpy
pytest pandas/tests/series/methods/test_describe.py::TestSeriesDescribe::test_numeric_result_dtype[complex]
```

--- a/pandas/tests/series/methods/test_describe.py
+++ b/pandas/tests/series/methods/test_describe.py
@@ -1,4 +1,7 @@
 import numpy as np
+import pytest
+
+from pandas.compat import is_numpy_dev
 
 from pandas.core.dtypes.common import (
     is_complex_dtype,
@@ -163,6 +166,12 @@ class TestSeriesDescribe:
             dtype = "complex128" if is_complex_dtype(any_numeric_dtype) else None
 
         ser = Series([0, 1], dtype=any_numeric_dtype)
+        if dtype == "complex128" and is_numpy_dev:
+            with pytest.raises(
+                TypeError, match=r"^a must be an array of real numbers$"
+            ):
+                ser.describe()
+            return
         result = ser.describe()
         expected = Series(
             [

